make/xml.cmake

```diff
@@ -37,3 +37,99 @@ function(XML_LINT)
 
 endfunction(XML_LINT)
 
+function(XML_SORT)
+  # ~~~
+  # XML_SORT(
+  # NAME
+  # FILE
+  # OUTPUT
+  # )
+  set(oneValueArgs NAME FILE OUTPUT)
+  cmake_parse_arguments(
+    XML_SORT
+    ""
+    "${oneValueArgs}"
+    ""
+    ${ARGN}
+    )
+
+  set(XML_SORT_XSL ${symbiflow-arch-defs_SOURCE_DIR}/common/xml/xmlsort.xsl)
+  set(
+    XML_SORT_INPUT ${CMAKE_CURRENT_BINARY_DIR}/${XML_SORT_FILE}
+  )
+  get_file_target(XML_SORT_INPUT_TARGET ${XML_SORT_FILE})
+  get_target_property(INCLUDE_FILES ${XML_SORT_INPUT_TARGET} INCLUDE_FILES)
+  set(DEPS "")
+  foreach(SRC ${INCLUDE_FILES})
+    append_file_dependency(DEPS ${SRC})
+  endforeach()
+  set(XML_SORT_OUTPUT ${CMAKE_CURRENT_BINARY_DIR}/${XML_SORT_OUTPUT})
+
+  get_target_property_required(XSLTPROC env XSLTPROC)
+  get_target_property(XSLTPROC_TARGET env XSLTPROC_TARGET)
+
+  add_custom_command(
+    OUTPUT ${XML_SORT_OUTPUT}
+    DEPENDS
+      ${XML_SORT_XSL}
+      ${XML_SORT_INPUT}
+      ${XML_SORT_INPUT_TARGET}
+      ${DEPS}
+      ${XSLTPROC} ${XSLTPROC_TARGET}
+    COMMAND
+      ${CMAKE_COMMAND} -E make_directory
+      ${CMAKE_CURRENT_BINARY_DIR}/${OUT_DEVICE_DIR}
+    COMMAND
+      ${XSLTPROC}
+      --nomkdir
+      --nonet
+      --xinclude
+      --output ${XML_SORT_OUTPUT} ${XML_SORT_XSL} ${XML_SORT_INPUT}
+  )
+  add_custom_target(
+    ${XML_SORT_NAME}
+    DEPENDS ${XML_SORT_OUTPUT}
+  )
+endfunction(XML_SORT)
+
+function(DIFF)
+  # ~~~
+  # DIFF(
+  # NAME
+  # GOLDEN
+  # ACTUAL
+  # )
+  set(oneValueArgs NAME GOLDEN ACTUAL)
+  cmake_parse_arguments(
+    DIFF
+    ""
+    "${oneValueArgs}"
+    ""
+    ${ARGN}
+    )
+
+  set(
+    DIFF_FILE_A ${CMAKE_CURRENT_BINARY_DIR}/${DIFF_GOLDEN}
+  )
+  set(
+    DIFF_FILE_B ${CMAKE_CURRENT_BINARY_DIR}/${DIFF_ACTUAL}
+  )
+  get_file_target(DIFF_FILE_A_TARGET ${DIFF_FILE_A})
+  get_file_target(DIFF_FILE_B_TARGET ${DIFF_FILE_B})
```

Comment:
Need to call get_file_location(DIFF_FILE_B_LOCATION ${DIFF_FILE_B})

Reply:
Done.